🌟 **Exercise 2** : Building a Simple Perceptron Decision System

You will manually create a simple perceptron-based decision system to determine whether you should go outside based on two inputs:

Temperature (°F)
Rainy (Yes = 1, No = 0)
You will assign weights, compute the weighted sum, apply an activation function, and determine the final decision.

Temperature weight = 0.6
Rain weight = 0.4
Bias = 2
Compute the weighted sum using the formula:

Weighted Sum=(Temperature×0.6)+(Rain×0.4)+Bias
Weighted Sum=(Temperature×0.6)+(Rain×0.4)+Bias
Apply a Step Activation Function:

If Weighted Sum > 20, output 1 (Yes, go outside)
If Weighted Sum ≤ 20, output 0 (No, stay inside)

### **1. Calculate the output for the following conditions:**

**Case 1: Temperature = 70°F, Rain = 0 (No)**

In [1]:
temp = 70
rain = 0

temp_weight = 0.6
rain_weight = 0.4
bias = 2

# Compute the weighted sum
weighted_sum = (temp * temp_weight) + (rain * rain_weight) + bias
print(weighted_sum)

# Activation function
if weighted_sum > 20:
    print("Yes, go outside")
else:
    print("No, stay inside")

44.0
Yes, go outside


**Case 2: Temperature = 50°F, Rain = 1 (Yes)**

In [2]:
temp = 50
rain = 1

temp_weight = 0.6
rain_weight = 0.4
bias = 2

# Compute the weighted sum
weighted_sum = (temp * temp_weight) + (rain * rain_weight) + bias
print(weighted_sum)

# Activation function
if weighted_sum > 20:
    print("Yes, go outside")
else:
    print("No, stay inside")

32.4
Yes, go outside


### **2. Interpret the results: Did the perceptron suggest going outside in both cases? Why or why not?**

Yes, the perceptron suggests going outside in both cases because the decision rule is based solely on whether the weighted sum is greater than 20. In both scenarios, even in case 2 where it is raining, the weighted sum (32.4) is still above the threshold. Therefore, the perceptron outputs 1, meaning "yes, go outside."